core/src/main/java/org/apache/iceberg/util/ThreadPools.java

```diff
@@ -61,6 +55,16 @@ public static ExecutorService getWorkerPool() {
     return WORKER_POOL;
   }
 
+  public static ExecutorService newWorkerPool(String nameFormat) {
```

Comment:
In other places, we use a name prefix and add `"-%d"` to that prefix so that we can guarantee it is there.

Reply:
And I think this should also accept a parallelism argument so that it is easily configurable.